src/Compilers/VisualBasic/Test/Semantic/Diagnostics/OperationAnalyzerTests.vb

```diff
@@ -415,5 +415,254 @@ End Class
+                                            Diagnostic(SymbolCouldHaveMoreSpecificTypeAnalyzer.FieldCouldHaveMoreSpecificTypeDescriptor.Id, "fi").WithArguments("Private fi As Object", "Integer").WithLocation(61, 13),
+                                            Diagnostic(SymbolCouldHaveMoreSpecificTypeAnalyzer.FieldCouldHaveMoreSpecificTypeDescriptor.Id, "fk").WithArguments("Private fk As Middle", "Derived").WithLocation(63, 13),
+                                            Diagnostic(SymbolCouldHaveMoreSpecificTypeAnalyzer.FieldCouldHaveMoreSpecificTypeDescriptor.Id, "fia").WithArguments("Private fia As Object", "IMiddle").WithLocation(72, 13),
+                                            Diagnostic(SymbolCouldHaveMoreSpecificTypeAnalyzer.FieldCouldHaveMoreSpecificTypeDescriptor.Id, "fic").WithArguments("Private fic As Object", "IBase1").WithLocation(73, 13),
+                                            Diagnostic(SymbolCouldHaveMoreSpecificTypeAnalyzer.FieldCouldHaveMoreSpecificTypeDescriptor.Id, "fid").WithArguments("Private fid As IBase1", "IDerived").WithLocation(74, 13),
+                                            Diagnostic(SymbolCouldHaveMoreSpecificTypeAnalyzer.FieldCouldHaveMoreSpecificTypeDescriptor.Id, "fie").WithArguments("Private fie As IBase1", "IMiddle").WithLocation(75, 13),
+                                            Diagnostic(SymbolCouldHaveMoreSpecificTypeAnalyzer.FieldCouldHaveMoreSpecificTypeDescriptor.Id, "fig").WithArguments("Private fig As Object", "IBase1").WithLocation(77, 13),
+                                            Diagnostic(SymbolCouldHaveMoreSpecificTypeAnalyzer.FieldCouldHaveMoreSpecificTypeDescriptor.Id, "fik").WithArguments("Private fik As IMiddle", "IDerived").WithLocation(79, 13),
+                                            Diagnostic(SymbolCouldHaveMoreSpecificTypeAnalyzer.FieldCouldHaveMoreSpecificTypeDescriptor.Id, "fim").WithArguments("Private fim As IBase2", "IDerived").WithLocation(83, 13))
+        End Sub
+
+        <Fact>
+        Public Sub ValueContextsVisualBasic()
+            Dim source = <compilation>
+                             <file name="c.vb">
+                                 <![CDATA[
+Class C
+    Public Sub M0(Optional a As Integer = 16, Optional b As Integer = 17, Optional c As Integer = 18)
+    End Sub
+
+    Public F1 As Integer = 16
+    Public F2 As Integer = 17
+    Public F3 As Integer = 18
+
+    Public Sub M1()
+        M0(16, 17, 18)
+        M0(f1, f2, f3)
+        M0()
+    End Sub
+End Class
+
+Enum E
+    A = 16
+    B
+    C = 17
+    D = 18
+End Enum
+
+Class C1
+    Public Sub New (a As Integer, b As Integer, c As Integer)
+    End Sub
+
+    Public F1 As C1 = New C1(c:=16, a:=17, b:=18)
+    Public F2 As New C1(16, 17, 18)
+    Public F3(16) As Integer
+    Public F4(17) As Integer                          ' The upper bound specification is not presently treated as a code block. This is suspect.
```

Comment:
This ought to be detected and presently is not. The logic to find code blocks in declarations is not finding array upper bound specifications. This issue is outside the scope of this change set.